### PR TITLE
Improvement in BatteryBar and CPUTemperature

### PR DESCRIPTION
**Pull request by vanjo9800**
I have designed a convenient way to see the remaining time of the battery when still seeing the icon. I have also made it easier to access the CPU temperature directly from the configuration using $temp$.

### CODE REVIEW

File: src/System/Taffybar/Text/CPUMonitor.hs [hunk 1]
```diff
@@ -18,12 +22,18 @@ textCpuMonitorNew fmt period = do
...
       (userLoad, systemLoad, totalLoad) <- cpuLoad
+      cpuTemp <- getCPUTemp["cpu0"]
       let [userLoad', systemLoad', totalLoad'] = map (formatPercent.(*100)) [userLoad, systemLoad, totalLoad]
+      let [cpuTemp'] = map (showTemp) [cpuTemp]
       let template = ST.newSTMP fmt
       let template' = ST.setManyAttrib [ ("user", userLoad'),
                                          ("system", systemLoad'),
-                                         ("total", totalLoad') ] template
+                                         ("total", totalLoad'),
+					 ("temp",cpuTemp')] template
```
Comment (anarcat) [inline src/System/Taffybar/Text/CPUMonitor.hs]: documentation string needs to be updated in the usage. otherwise: nice job, i've been looking for something like this.

File: taffybar.cabal [hunk 2]
```diff
@@ -53,7 +53,8 @@ library
                  safe >= 0.3 && < 1,
                  split >= 0.1.4.2,
                  process >= 1.0.1.1,
-                 enclosed-exceptions >= 1.0.0.1
+                 enclosed-exceptions >= 1.0.0.1,
+		 gtk2hs-buildtools >= 0.13.0.0
```
Comment (anarcat) [inline taffybar.cabal]: why is this necessary?

**Comment by anarcat** [comment]
@vanjo9800 there was a new major release, so you'll need to reroll your changes to fix the conflicts here.

**Comment by IvanMalison** [comment]
@anarcat any interest in taking this over?

**Comment by anarcat** [comment]
i'm not sure i'm the right person to do this. all my Haskell experiments are cargo-culting and voodoo experiments... after much pain, i have arrived at this nasty piece of business:

```haskell
tempZoneWidget :: Int -> IO Widget
tempZoneWidget zone = do
  label <- pollingLabelNew "N/A C" 1.0 $ callback
  widgetShowAll label
  return label
  where
    path = "/sys/devices/virtual/thermal/thermal_zone" ++ show zone ++ "/temp"
    callback = do
        temp <- readFile $ path
        let tempn = (read temp) :: Int
        let tempd = (fromIntegral tempn) / 1000
            fmt :: Double -> String
            fmt celc = printf "zone%d: %f°C" zone celc
        return (fmt tempd)
```

... which is used, clunkily, as:

```haskell
main = do
  let
      temp0 = tempZoneWidget 0
      temp3 = tempZoneWidget 3
 defaultTaffybar defaultTaffybarConfig { startWidgets = [ pager ], endWidgets = [ temp0, temp3 ] }
```

surely there is a better way, particularly to get autodiscovery of the available temperatures, but I really got tired of this and stopped there...

suggestions?

**Comment by IvanMalison** [comment]
@anarcat @vanjo9800 see changes that were recently pushed to master as part of integrating #119

**Comment by anarcat** [comment]
i guess the problem i had with getCPUTemp is i had trouble turning it into a working widget... any suggestions?

**Comment by IvanMalison** [comment]
@anarcat If you just want a text widget it should be really easy

**Comment by IvanMalison** [comment]
Something like

``` haskell
getAverageTemp = fromMaybe "Error" . fmap (show . snd) <$> getMTuple
  where getMTuple = lookup "temp1_input" <$> getCPUTemperatures

cpuTemp = pollingLabelNew "" 15.0 getAverageTemp
```

probably won't compile, but you can figure out the details.

**Comment by IvanMalison** [comment]
@anarcat You can ping me on https://gitter.im/taffybar/Lobby if you have any more questions

**Comment by anarcat** [comment]
@IvanMalison yes, a text widget. assume I have no clue what i'm doing and the above is just something i cargo-culted and bruteforced until it worked. :p

**Comment by IvanMalison** [comment]
@anarcat did you try what I posted? You'll need to import Data.Maybe

**Comment by anarcat** [comment]
oops, sorry - i hadn't see the previous comment showing how to actually build the widget! :)

i need to deploy the latest source, i still haven't managed to do that just yet... sorry for the delay and confusion!

it looks like the code snippet you provided shows temperature for probe 1, is there a way to show all active probes?

**Comment by IvanMalison** [comment]
I believe that temp1 is the aggregated temperature. If you REALLY want all of the temperatures, something like:

```
getAllTemp = intercalate ", " . map (show . snd) <$> getCPUTemperatures

cpuTemp = pollingLabelNew "" 15.0 getAllTemp
```

that will just make the label be a comma separated list of the temperatures.

Not sure if that is what you want.
